zerver/migrations/0266_migrate_old_bot_messages.py

```diff
@@ -0,0 +1,57 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.conf import settings
+from django.db import migrations
+from django.db.backends.postgresql_psycopg2.schema import DatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+
+from typing import Any
+
+def fix_messages(apps: StateApps, schema_editor: DatabaseSchemaEditor) -> None:
+    UserProfile = apps.get_model('zerver', 'UserProfile')
+    Message = apps.get_model('zerver', 'Message')
+    Realm = apps.get_model('zerver', 'Realm')
+
+    try:
+        internal_realm = Realm.objects.get(string_id=settings.SYSTEM_BOT_REALM)
+    except Realm.DoesNotExist:
+        # Server not initialized, or no system bot realm. Either way, we shouldn't do anything.
+        return
+
+    def get_bot_by_delivery_email(email: str) -> Any:
+        return UserProfile.objects.select_related().get(delivery_email__iexact=email.strip(),
+                                                        realm=internal_realm)
+
+    notification_bot = get_bot_by_delivery_email(settings.NOTIFICATION_BOT)
+
+    def fix_messages_by_bot(bot_profile: Any) -> None:
+        Message.objects.filter(sender=bot_profile).update(sender=notification_bot)
+        Message.objects.filter(recipient=bot_profile.recipient).update(recipient=notification_bot.recipient)
+
+    new_user_bot_email = getattr(settings, 'NEW_USER_BOT', 'new-user-bot@zulip.com')
+    try:
+        new_user_bot = get_bot_by_delivery_email(new_user_bot_email)
+        fix_messages_by_bot(new_user_bot)
+        new_user_bot.delete()
```

Comment:
Do we still need to delete the personal `Recipient` objects?

Reply:
Good catch, thanks!